mailer.rst

```diff
@@ -0,0 +1,653 @@
+
+Each library includes a :ref:`Flex recipe <flex-recipe>` that will add example configuration
+to your ``.env`` file. For example, suppose you want to use SendGrid. First,
+install it:
+
+.. code-block:: terminal
+
+    $ composer require symfony/symfony/sendgrid-mailer
+
+You'll now have a new line in your ``.env`` file that you can uncomment:
+
+.. code-block:: bash
+
+    # .env`
+
+    SENDGRID_KEY=
+    MAILER_DSN=smtp://$SENDGRID_KEY@sendgrid
+
+The``MAILER_DSN`` isn't a *real* SMTP address: it's a simple format that offloads
+most of the configuration work to mailer. The ``@sendgrid`` part of the address
+activates the SendGrid mailer library that you just installed, which knows all
+about how to deliver messages to SendGrid.
+
+The *only* part you need to change is to set ``SENDGRID_KEY`` to your key (in
+``.env`` or ``.env.local``).
+
+Each transport will have different environment variables that the library will use
+to configure the *actual* address and authentication for delivery. Some also have
+options that can be configured with query parameters on end of the ``MAILER_DSN`` -
+like ``?region=`` for Amazon SES. Some transports support sending via ``http``
+or ``smtp`` - both work the same, but ``http`` is recommended when available.
+
+Creating & Sending Messages
+---------------------------
+
+To send an email, autowire the mailer using
+:class:`Symfony\\Component\\Mailer\\MailerInterface` (service id ``mailer``)
+and create an :class:`Symfony\\Component\\Mime\\Email` object::
+
+    // src/Controller/MailerController.php
+    namespace App\Controller;
+
+    use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+    use Symfony\Component\Mailer\MailerInterface;
+    use Symfony\Component\Mime\Email;
+
+    class MailerController extends AbstractController
+    {
+        /**
+         * @Route("/email")
+         */
+        public function sendEmail(MailerInterface $mailer)
+        {
+            $email = (new Email())
+                ->from('hello@symfony.com')
+                ->to('you@example.com')
+                //->cc('cc@example.com')
+                //->bcc('bcc@example.com')
+                //->replyTo('fabien@symfony.com')
+                //->priority(Email::PRIORITY_HIGH)
+                ->subject('Time for Symfony Mailer!!!99')
+                ->text('Sending emails is fun again!')
+                ->html('<p>See Twig integration for better HTML integration!</p>');
+
+            $mailer->send($email);
+
+            // ...
+        }
+    }
+
+That's it! The message will be sent via whatever transport you configured.
+
+Email Addresses
+~~~~~~~~~~~~~~~
+
+All the methods that require email addresses (``from()``, ``to()``, etc.) accept
+both strings and objects::
+
+    // ...
+    use Symfony\Component\Mime\Address;
+    use Symfony\Component\Mime\NamedAddress;
+
+    $email = (new Email())
+        // email address as a simple string
+        ->from('fabien@symfony.com')
```

Comment:
Introducing a real email will spam people. It may be good to use example.com instead of real domain.